### ISSUE TITLE
FF 66 broke plugin

### ISSUE THREAD

**Issue by adi-** [bbarakaci/fixto#56]
Any clue what Firefox 66 broke in this plugin? I see an instant jump when plugin is activated with `fixto-fixed`.

**Comment by bbarakaci** [comment]
Pfff, not yet.

Release notes mention such a thing:

> Smoother scrolling: Scroll anchoring keeps content from jumping as images and ads load at the top of the page

https://www.mozilla.org/en-US/firefox/66.0/releasenotes/

I am observing it on the last example of the demo page: http://bbarakaci.github.io/fixto/dev2.html

**Comment by bbarakaci** [comment]
If firefox setting layout.css.scroll-anchoring.enabled set to false, it works as before.

As fixed positioned elements go out of the flow, a replacer element takes its place, right before setting it to fixed. This happens so fast that the user does not observe. But Firefox detects this change and just scrolls down as much as the height of the replacer element, creating the jumping effect.

**Comment by adi-** [comment]
Yeah, exactly... Can we do something about it? ;)

**Comment by bbarakaci** [comment]
Yesterday I invested a good amount of time with no results. I did the fixing bit in an animation frame which ensures replacing and fixing happens between two paints, but didn’t make a difference. Looks like fireworks does not check if the scroll amount changed or not, it reacts to the layout change. That replacer was the main magic of this plugin.

I suggest turning on the useNativeSticky option as the first step and hope your requirements are still met.

For the long run I suggest dropping this plugin and going with css position sticky or finding another library.

I did not have time to maintain this plugin for some years and I am afraid I won’t have time in the future.

Sorry for any inconvenience.

**Comment by adi-** [comment]
> I suggest turning on the useNativeSticky option as the first step and hope your requirements are still met.

Yeah, this is what I am going to do. However, I would miss auto class addition which styled my navbar when scrolled :(

> For the long run I suggest dropping this plugin and going with css position sticky or finding another library.
> I did not have time to maintain this plugin for some years and I am afraid I won’t have time in the future.

I understand.

**Comment by bbarakaci** [comment]
A dirty solution will be listening to scroll events yourself and adding the class when scrollTop is greater than x, and removing on else case.

A better solution will be using the intersection observer but it will be trickier.

**Comment by adi-** [comment]
> A dirty solution will be listening to scroll events yourself and adding the class when scrollTop is greater than x, and removing on else case.

Good idea! Thanx!
